Change modify_disabled to create_disabled and delete_disabled

tung2744 opened the pull request:
ref DEV-1226

louischan-oursky commented on pkg/api/oauthrelyingparty/provider.go:
```diff
@@ -32,15 +32,26 @@ func (c ProviderConfig) ClientID() string {
+	return modify_disabled
+}
+func (c ProviderConfig) DeleteDisabled() bool {
+	modify_disabled, _ := c["delete_disabled"].(bool)
```
ModifyDisabled is removed. But for the following case, it should be read.

The actual config
```yaml
modify_disabled: true
```
This project has configured modify_disabled=true for a particular provider.

tung2744 replied:
@louischan-oursky I think this part should be handled by the implementation of the provider.

Something like:

```go
func (ProviderABC) SetDefaults(cfg oauthrelyingparty.ProviderConfig) {
	deprecatedModifyDisabled, ok := cfg["modify_disabled"].(bool)
	if ok {
		cfg.SetDefaultsUpdateDisabled(deprecatedModifyDisabled)
	}
}
```

louischan-oursky replied:
According to our offline discussion, we will remove `Alias()`, `SetDefaultsModifyDisabledFalse()`, and `ModifyDisabled()`

tung2744 replied:
I pushed a new commit for this change, but I am still working on the authgear-server changes.

tung2744 committed:
Simplify oauthrelyingparty provider

- Implementation of providers should not aware of fields that are used by authgear only

tung2744 committed:
Add GetJSONSchema to Provider
